hw/dv/dpi/gpiodpi/gpiodpi.c

```diff
@@ -21,46 +21,108 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+// This file does a lot of bit setting and getting; these macros are intended to
+// make that a little more readable.
+#define GET_BIT(word, bit_idx) (((word) >> bit_idx) & 1)
+#define SET_BIT(word, bit_idx) (word |= (1 << bit_idx))
+#define CLR_BIT(word, bit_idx) (word &= ~(1 << bit_idx))
```

Comment:
Wrap `word` and `bit_idx` uses in parens too.

Reply:
I hate macros.